backend/planty/dashboard/views.py

```diff
@@ -290,3 +295,226 @@ def post(self, request: Request):
+
+                'num_selected': instruction.num_selected
+            }
+            desc_suggested_instructions_json.append(instruction_json)
+
+        return Response(desc_suggested_instructions_json, status=status.HTTP_200_OK)
+
+
+class SelectInstructionView(APIView):
+    def get(self, request: Request):
+        user: User = request.user
+        serializer = InstructionSelectSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        try:
+            instruction: Instruction = Instruction.objects.get(pk=data['id'])
+        except Model.DoesNotExist:
+            return Response(data={
+                'id': ['Plant with the given ID does not exist']
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if instruction.user != user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if not instruction.public:
+            return Response(data={
+                'id:': ['Instruction with this ID is not public. Cannot select non public instructions.']
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        instruction.num_selected += 1
+
+        # creating public instruction
+
+        shared_instruction: Instruction = Instruction.objects.create(
+            id=uuid4(),
+            name=instruction.name,
+
+            user=user,
+            species=data['species'],
+
+            watering=data['watering'],
+            insolation=data['insolation'],
+            fertilizing=data['fertilizing'],
+
+            public=False
+        )
+
+        shared_instruction.save()
+
+        return Response(status=status.HTTP_201_CREATED)
+
+
+class ShareInstructionView(APIView):
+    def post(self, request: Request):
+        user: User = request.user
+        serializer = InstructionShareSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        try:
+            instruction: Instruction = Instruction.objects.get(pk=data['id'])
+        except Model.DoesNotExist:
+            return Response(data={
+                'id': ['Plant with the given ID does not exist']
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if instruction.user != user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if instruction.public:
+            return Response(data={
+                'id:': ['Instruction with this ID is public. Cannot share public instructions.']
+            }, status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+        # creating public instruction
+
+        shared_instruction: Instruction = Instruction.objects.create(
+            id=uuid4(),
+            name=instruction.name,
+
+            user=user,
+            species=data['species'],
```

Comment:
Przecież nie ma tych danych w request'ie. Tutaj po prostu trzeba to skopiować z `instruction`, czyli `species=instruction.species` i td.

Reply:
racja, naprawione